src/components/optimal-promotion-item/optimal-promotion-item.tsx

```diff
@@ -35,13 +35,9 @@ export const OptimalPromotionItem: FC<Props> = ({
   const partnersPromotionEnabled = useIsPartnersPromoEnabledSelector();
   const disablePromotionAfterConfirmation = useDisablePromotionAfterConfirmation();
 
-  if (!partnersPromotionEnabled) {
-    return null;
-  }
-
-  if (isEmptyPromotion(partnersPromotion)) {
-    onEmptyPromotionReceived?.();
+  useEffect(() => void onEmptyPromotionReceived?.(), []);
```

Comment:
```suggestion
  useEffect(() => isEmptyPromotion(partnersPromotion) && onEmptyPromotionReceived?.(), [partnersPromotion, onEmptyPromotionReceived]);
```
Or something like this